functions.cmake

```diff
@@ -247,6 +251,9 @@ endfunction()
 function(_add_library)
     if(NOT WIN32)
       add_library(${ARGV} ${VERSION_FILE_PATH})
+      if (HAVE_LTO AND CLR_CMAKE_BUILD_COREBUNDLE AND ${ARGV1} EQUAL "STATIC")
+        target_compile_options(${ARGV0} PRIVATE -flto)
```

Comment:
Why do we this only for static libraries?

Reply:
I was originally trying to limit it to just what would matter for corebundle. Revisiting though I realized that I actually used `EQUAL` instead of `STREQUAL`, so this isn't actually doing anything...

I'm just entirely removing this for now and will look at enabling LTO for more as a follow-up item/change.